packages/example-contracts/contracts/cross-chain-counter/CrossChainCounter.sol

```diff
@@ -2,43 +2,29 @@
 pragma solidity 0.8.7;
 
 import "@openzeppelin/contracts/access/Ownable.sol";
+import "@zetachain/protocol-contracts/contracts/ZetaInteractor.sol";
 import "@zetachain/protocol-contracts/contracts/interfaces/ZetaInterfaces.sol";
 
-contract CrossChainCounter is Ownable, ZetaReceiver {
-    bytes32 public constant CROSS_CHAIN_INCREMENT_MESSAGE = keccak256("CROSS_CHAIN_INCREMENT");
+interface CrossChainCounterErrors {
+    error InvalidMessageType();
 
-    address public connectorAddress;
-    ZetaConnector internal connector;
+    error DecrementOverflow();
+}
 
-    uint256 internal immutable currentChainId;
-    uint256 internal _crossChainId;
-    bytes internal _crossChainAddress;
+contract CrossChainCounter is ZetaInteractor, ZetaReceiver, CrossChainCounterErrors {
+    bytes32 public constant CROSS_CHAIN_INCREMENT_MESSAGE = keccak256("CROSS_CHAIN_INCREMENT");
 
     mapping(address => uint256) public counter;
 
-    constructor(address connectorAddress_) {
-        currentChainId = block.chainid;
-
-        connectorAddress = connectorAddress_;
-        connector = ZetaConnector(connectorAddress_);
-    }
-
-    function setCrossChainAddress(bytes calldata ccAddress) public onlyOwner {
-        _crossChainAddress = ccAddress;
-    }
-
-    function setCrossChainId(uint256 ccId) public onlyOwner {
-        _crossChainId = ccId;
-    }
+    constructor(address connectorAddress_) ZetaInteractor(connectorAddress_) {}
 
-    function crossChainCount() external {
-        require(_crossChainAddress.length != 0, "Cross-chain address is not set");
-        require(_crossChainId != 0, "Cross-chain id is not set");
+    function crossChainCount(uint256 crossChainId) external {
```

Comment:
Can we change this one to `destinationChainId`? I think it'll make the API more clear.